src/Build/Instance/ProjectInstance.cs

```diff
@@ -2597,6 +2633,28 @@ private void CreateTargetsSnapshot(Evaluation.Project.Data data)
             _targets = CreateCloneDictionary(data.Targets);
         }
 
+        /// <summary>
+        /// Create various import snapshots
+        /// </summary>
+        private void CreateImportSnapshot(Evaluation.Project.Data data)
+        {
+            _imports = new List<string>(data.ImportClosure.Count);
+            foreach (var resolvedImport in data.ImportClosure)
```

Comment:
I think you could condense these loops into one.

Reply:
They're different collections with (potentially) different content.

Reply:
One way:
Make Import a set instead of a list. It's not supposed to have duplicates anyway; this enforces that.
Alternative:
If the duplicates list is sorted, check if the next element is the same as the last one and don't add it to _imports if so.

Reply:
This is following the implementation in Project. Deviating a lot between the two seems like it'll cause more problems than solve any.

The List vs Set thing is enforced by the Evaluator. This is merely being reported to a public-facing API.

Reply:
I think you want it a List to preserve information about the import order. Single loop does not make sense to me as the number of iterations can differ, and this code is not in a hot path so intense that would warrant such optimizations.